WebServices/AzureNotificationHub/NotificationHubSample/NotificationHubSample.iOS/AppDelegate.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Foundation;
+using UIKit;
+using UserNotifications;
+using WindowsAzure.Messaging;
+
+namespace NotificationHubSample.iOS
+{
+    [Register("AppDelegate")]
+    public partial class AppDelegate : global::Xamarin.Forms.Platform.iOS.FormsApplicationDelegate
+    {
+        private SBNotificationHub Hub { get; set; }
+
+        public override bool FinishedLaunching(UIApplication app, NSDictionary options)
+        {
+            global::Xamarin.Forms.Forms.Init();
+            LoadApplication(new App());
+
+            base.FinishedLaunching(app, options);
+
+            // register for remote notifications based on system version
+            if (UIDevice.CurrentDevice.CheckSystemVersion(10, 0))
+            {
+                UNUserNotificationCenter.Current.RequestAuthorization(UNAuthorizationOptions.Alert | 
+                    UNAuthorizationOptions.Sound |
+                    UNAuthorizationOptions.Sound,
+                    (granted, error) =>
+                        {
+                            if (granted)
+                                InvokeOnMainThread(UIApplication.SharedApplication.RegisterForRemoteNotifications);
+                        });
+            }
+            else if (UIDevice.CurrentDevice.CheckSystemVersion(8, 0))
```

Comment:
While this isn't doing any harm, I'm just saying that we made a decision, a while ago, to support back to iOS 9 in our docs/samples. That may now be iOS 10, as there was talk of supporting the last 3 versions.

Reply:
I'm going to leave this in just because it feels like a bad gotcha. Registering is OS-version-dependent and you have to test this on real devices. So if the dev only has an older physical device it will fail silently and they will have no idea why.